### PR TITLE
[INLONG-8839][Sort] Add audit metric in starrocks connector on flink 1.15

### LINKED ISSUE #8839: [Feature] Add audit metric in starrocks connector on flink 1.15
### Description

[Feature] Add audit metric in starrocks connector on flink 1.15 

### Use case

[Feature] Add audit metric in starrocks connector on flink 1.15 

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by EMsnap**
### Prepare a Pull Request

- Fixes #8839 

### Motivation

Add audit metric in starrocks connector on flink 1.15

### Modifications

Add audit metric in starrocks connector on flink 1.15

1、modify end to end test to cope with starrocks trasnaction load
2、add starrocks sink function v1 to support starrocks version before v2.4
3、add schema utils to extract data time 

### Verifying this change

Run Iceberg -> starrocks

Check the audit data 

image

### CODE REVIEW

File: inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/starrocks/pom.xml [hunk 1]
```diff
@@ -45,6 +45,10 @@
             <artifactId>flink-connector-starrocks</artifactId>
             <version>${starrocks-connector.version}_flink-${flink.starrocks.version}</version>
         </dependency>
+        <dependency>
```
Comment (yunqingmoswu) [inline inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/starrocks/pom.xml:48]: It is recommend to use jackson or google json.

File: inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/starrocks/src/main/java/org/apache/inlong/sort/starrocks/table/sink/table/StarRocksDynamicSinkFunctionV2.java [hunk 2]
```diff
@@ -0,0 +1,370 @@
...
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+
+public class StarRocksDynamicSinkFunctionV2<T> extends StarRocksDynamicSinkFunctionBase<T> {
```
Comment (yunqingmoswu) [inline inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/starrocks/src/main/java/org/apache/inlong/sort/starrocks/table/sink/table/StarRocksDynamicSinkFunctionV2.java:72]: Please add some comments.

File: inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/starrocks/src/main/java/org/apache/inlong/sort/starrocks/table/sink/table/StarRocksDynamicTableSink.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+import com.starrocks.connector.flink.table.sink.StarRocksDynamicSinkFunctionBase;
+import com.starrocks.connector.flink.table.sink.StarRocksSinkOptions;
+import org.apache.flink.api.common.typeinfo.TypeInformation;
+import org.apache.flink.table.api.TableSchema;
+import org.apache.flink.table.connector.ChangelogMode;
+import org.apache.flink.table.connector.sink.DynamicTableSink;
+import org.apache.flink.table.connector.sink.SinkFunctionProvider;
+import org.apache.flink.table.data.RowData;
+
+public class StarRocksDynamicTableSink implements DynamicTableSink {
```
Comment (yunqingmoswu) [inline inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/starrocks/src/main/java/org/apache/inlong/sort/starrocks/table/sink/table/StarRocksDynamicTableSink.java:30]: Please add some comments

File: inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/starrocks/pom.xml (hunk 1 above)
Comment (EMsnap) [reply inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/starrocks/pom.xml:48]: Thanks, it's tested in official starrocks connector, will do in the next version

**Review by yunqingmoswu — APPROVED** [review]

**Review by e-mhui — APPROVED** [review]

**Review by gong — APPROVED** [review]

**Review by vernedeng — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**